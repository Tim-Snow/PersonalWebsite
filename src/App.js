import React from 'react'

import Profile from './components/Profile'
import AboutMe from './components/AboutMe'
import Activity from './components/Activity'

function App() {
  return (
    <div className="App">
      <Profile />
      <div style={styles.bottomSection}>
        <AboutMe />
        <Activity />
      </div>
    </div>
  )
}

const styles = {
  bottomSection: {
    display: 'flex',
    flex: 1,
    flexDirection: 'row',
    flexWrap: 'wrap',
    justifyContent: 'space-around',
  },
}

export default App
