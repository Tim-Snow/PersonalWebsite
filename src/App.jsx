import React from 'react';

import Profile from './components/Profile';
import AboutMe from './components/AboutMe';
import Activity from './components/Activity';
import Portfolio from './components/Portfolio';

import { background } from './constants/styles';

function App() {
  return (
    <div style={styles.container}>
      <Profile />
      <div style={styles.splitSection}>
        <AboutMe style={styles.about} />
        <Activity style={styles.activity} />
      </div>
      <Portfolio />
    </div>
  );
}

const styles = {
  container: {
    backgroundColor: background,
  },
  splitSection: {
    display: 'flex',
    flex: 1,
    flexDirection: 'row',
    flexWrap: 'wrap',
    justifyContent: 'space-around',
  },
  about: {
    flex: 2,
  },
  activity: {
    flex: 1,
  },
};

export default App;
