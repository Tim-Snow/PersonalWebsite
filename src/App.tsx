import React from 'react';

import Profile from 'features/Profile';
import AboutMe from 'features/AboutMe';
import Activity from 'features/Activity';
import Portfolio from 'features/Portfolio';
import NavBar from 'features/NavBar';

import { background } from 'constants/styles';

const styles = {
  container: {
    backgroundColor: background,
    paddingTop: 56,
  },
  splitSection: {
    display: 'flex',
    flex: 1,
    flexDirection: 'row',
    flexWrap: 'wrap',
    justifyContent: 'space-around',
  },
} as { [key: string]: React.CSSProperties };

export default function App() {
  return (
    <div style={styles.container} id="home">
      <NavBar />
      <Profile />
      <Portfolio />
      <div style={styles.splitSection}>
        <AboutMe />
        <Activity />
      </div>
    </div>
  );
}
