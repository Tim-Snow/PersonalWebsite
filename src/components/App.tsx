import * as React from 'react'
import { connect } from 'react-redux'

import Game from './game'
import Normal from './normal'
import Header from './header'

import * as Page from '../constants/pages'

interface Props {
  type: string
}

class App extends React.Component<Props> {
  public render() {
    return (
      <div style={{ textAlign: 'center', margin: '20px' }}>
        <Header />
        {(this.props.type === Page.GAME && <Game />) || <Normal />}
      </div>
    )
  }
}

function mapStateToProps(state: any) {
  return {
    type: state.navigation.type,
  }
}

export default connect(mapStateToProps)(App)
