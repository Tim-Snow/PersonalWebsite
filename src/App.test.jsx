import React from 'react'
import ReactDOM from 'react-dom'
import App from './App'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16.3'

beforeAll(() => {
  Enzyme.configure({ adapter: new Adapter() })
})

it('looks sick', () => {
  const div = document.createElement('div')
  ReactDOM.render(<App />, div)
  ReactDOM.unmountComponentAtNode(div)
  expect(shallow(<App />)).toMatchSnapshot()
})
