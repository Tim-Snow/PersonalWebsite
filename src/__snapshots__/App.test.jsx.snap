// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`looks sick 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Profile />,
        <div
          style={
            Object {
              "display": "flex",
              "flex": 1,
              "flexDirection": "row",
              "flexWrap": "wrap",
              "justifyContent": "space-around",
            }
          }
        >
          <AboutMe
            style={
              Object {
                "flex": 2,
              }
            }
          />
          <Activity
            style={
              Object {
                "flex": 1,
              }
            }
          />
        </div>,
        <Portfolio />,
        <Contact />,
      ],
      "style": Object {
        "backgroundColor": "#335",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <AboutMe
              style={
                Object {
                  "flex": 2,
                }
              }
            />,
            <Activity
              style={
                Object {
                  "flex": 1,
                }
              }
            />,
          ],
          "style": Object {
            "display": "flex",
            "flex": 1,
            "flexDirection": "row",
            "flexWrap": "wrap",
            "justifyContent": "space-around",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "style": Object {
                "flex": 2,
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "style": Object {
                "flex": 1,
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Profile />,
          <div
            style={
              Object {
                "display": "flex",
                "flex": 1,
                "flexDirection": "row",
                "flexWrap": "wrap",
                "justifyContent": "space-around",
              }
            }
          >
            <AboutMe
              style={
                Object {
                  "flex": 2,
                }
              }
            />
            <Activity
              style={
                Object {
                  "flex": 1,
                }
              }
            />
          </div>,
          <Portfolio />,
          <Contact />,
        ],
        "style": Object {
          "backgroundColor": "#335",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <AboutMe
                style={
                  Object {
                    "flex": 2,
                  }
                }
              />,
              <Activity
                style={
                  Object {
                    "flex": 1,
                  }
                }
              />,
            ],
            "style": Object {
              "display": "flex",
              "flex": 1,
              "flexDirection": "row",
              "flexWrap": "wrap",
              "justifyContent": "space-around",
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "style": Object {
                  "flex": 2,
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "style": Object {
                  "flex": 1,
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenThreeAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
